src/chains/MegaEth/MetaTxn.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.25;
+
+import {MegaEthMixin} from "./Common.sol";
+import {SettlerMetaTxn} from "../../SettlerMetaTxn.sol";
+
+import {IERC20} from "@forge-std/interfaces/IERC20.sol";
+import {ISignatureTransfer} from "@permit2/interfaces/ISignatureTransfer.sol";
+import {ISettlerActions} from "../../ISettlerActions.sol";
+
+// Solidity inheritance is stupid
+import {SettlerAbstract} from "../../SettlerAbstract.sol";
+import {SettlerBase} from "../../SettlerBase.sol";
+import {AbstractContext} from "../../Context.sol";
+
+/// @custom:security-contact security@0x.org
+contract MegaEthSettlerMetaTxn is SettlerMetaTxn, MegaEthMixin {
+    constructor(bytes20 gitCommit) SettlerBase(gitCommit) {}
+
+    function _dispatchVIP(uint256 action, bytes calldata data, bytes calldata sig)
+        internal
+        virtual
+        override
+        DANGEROUS_freeMemory
+        returns (bool)
+    {
+        // This does not make use of `super._dispatchVIP`. This chain's Settler is extremely
+        // stripped-down and has almost no capabilities
```

Comment:
shouldn't this one be actually calling `super._dispatch` so the `MegaEthMixin` dispatch gets executed and `MegaEthSettlerMetaTxn ` can do `BASIC`